reddit_service_websockets/socketserver.py

```diff
@@ -106,20 +111,48 @@ def start_response(self, status, headers, exc_info=None):
 
 
 class SocketServer(object):
-    def __init__(self, metrics, dispatcher, mac_secret, ping_interval):
+    def __init__(self, metrics,
+                 dispatcher,
+                 mac_secret,
+                 ping_interval,
+                 admin_auth,
+                 conn_shed_rate,
+    ):
         self.metrics = metrics
         self.dispatcher = dispatcher
         self.signer = MessageSigner(mac_secret)
         self.ping_interval = ping_interval
+        self.admin_auth = admin_auth
+        self.shed_rate_per_sec = conn_shed_rate
         self.status_publisher = None
+        self.quiesced = False
+        self.connections = set()
 
     def __call__(self, environ, start_response):
         path_info = environ["PATH_INFO"]
+
+        if path_info == '/quiesce':
```

Comment:
I think that actually makes sense here since we're changing state with the first request.